pkg/workload/workload.go

```diff
@@ -652,7 +652,7 @@ func PropagateResourceRequests(w *kueue.Workload, info *Info) bool {
 		match := true
 		for idx := range w.Status.ResourceRequests {
 			if w.Status.ResourceRequests[idx].Name != info.TotalRequests[idx].Name ||
-				!maps.Equal(w.Status.ResourceRequests[idx].Resources, info.TotalRequests[idx].Requests.ToResourceList()) {
+				!IsResourceListEqual(w.Status.ResourceRequests[idx].Resources, info.TotalRequests[idx].Requests.ToResourceList()) {
```

Comment:
```suggestion
				!equality.Semantic.DeepEqual(w.Status.ResourceRequests[idx].Resources, info.TotalRequests[idx].Requests.ToResourceList()) {
```

The kubernetes `Semantic.DeepEqual` allows us to verify same thing. You can try the following codes.

```go
func main() {
	a := corev1.ResourceList{"cpu": *resource.NewScaledQuantity(10, 0)}
	b := corev1.ResourceList{"cpu": *resource.NewScaledQuantity(10000, -3)}
	fmt.Println(equality.Semantic.DeepEqual(a, b))
}
```

Reply:
ah, nice